Subject: [PATCH] login page.
---
Index: src/main/resources/templates/index.html
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/resources/templates/index.html b/src/main/resources/templates/index.html
--- a/src/main/resources/templates/index.html	(revision 3bb14e1b120aa5df4e884ed09ec2656b3eddaca6)
+++ b/src/main/resources/templates/index.html	(date 1713086937382)
@@ -22,7 +22,7 @@
             <span id="menu-btn"></span>
 
             <div class="row">
-                <div th:block th:include="~{/fragment/headerFragment::siteMenu}"></div>
+                <div th:block th:insert="~{/fragment/headerFragment::siteMenu}"></div>
             </div>
         </div>
     </header>
Index: src/main/java/com/example/cargo/endpoint/UserController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/example/cargo/endpoint/UserController.java b/src/main/java/com/example/cargo/endpoint/UserController.java
--- a/src/main/java/com/example/cargo/endpoint/UserController.java	(revision 3bb14e1b120aa5df4e884ed09ec2656b3eddaca6)
+++ b/src/main/java/com/example/cargo/endpoint/UserController.java	(date 1713089390108)
@@ -2,7 +2,6 @@
 
 import com.example.cargo.entity.User;
 import com.example.cargo.entity.enums.UserRole;
-import com.example.cargo.repository.UserRepository;
 import com.example.cargo.security.SpringUser;
 import com.example.cargo.service.UserService;
 import jakarta.validation.Valid;
@@ -15,8 +14,6 @@
 import org.springframework.validation.BindingResult;
 import org.springframework.web.bind.annotation.*;
 
-import java.util.Optional;
-
 @Controller
 @RequestMapping("/user")
 public class UserController {
Index: src/main/java/com/example/cargo/configuration/SecurityConfig.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/example/cargo/configuration/SecurityConfig.java b/src/main/java/com/example/cargo/configuration/SecurityConfig.java
--- a/src/main/java/com/example/cargo/configuration/SecurityConfig.java	(revision 3bb14e1b120aa5df4e884ed09ec2656b3eddaca6)
+++ b/src/main/java/com/example/cargo/configuration/SecurityConfig.java	(date 1713088981780)
@@ -8,7 +8,7 @@
 import org.springframework.security.authentication.dao.DaoAuthenticationProvider;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
-import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
 import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.security.web.SecurityFilterChain;
 
@@ -23,24 +23,23 @@
 
     @Bean
     public SecurityFilterChain configure(HttpSecurity httpSecurity) throws Exception {
-        httpSecurity.csrf().disable()
-                .authorizeHttpRequests()
-                .requestMatchers("/").permitAll()
-                .requestMatchers("/user/registration").permitAll()
-                .requestMatchers("/loginPage").permitAll()
-                .requestMatchers("/gallery").permitAll()
-                .requestMatchers("/about").permitAll()
-                .requestMatchers("/calculate").permitAll()
-                .requestMatchers("/contact").permitAll()
-                .anyRequest().permitAll()
-                .and()
-                .formLogin()
-                .loginPage("/loginPage")
-                .loginProcessingUrl("/login")
-                .defaultSuccessUrl("/loginSuccess")
-                .and()
-                .logout()
-                .logoutSuccessUrl("/");
+        httpSecurity.csrf(AbstractHttpConfigurer::disable)
+                .authorizeHttpRequests(authorizationManagerRequestMatcherRegistry
+                        -> authorizationManagerRequestMatcherRegistry
+                        .requestMatchers("/").permitAll()
+                        .requestMatchers("/user/registration").permitAll()
+                        .requestMatchers("/loginPage").permitAll()
+                        .requestMatchers("/gallery").permitAll()
+                        .requestMatchers("/about").permitAll()
+                        .requestMatchers("/calculate").permitAll()
+                        .requestMatchers("/contact").permitAll()
+                        .anyRequest().permitAll())
+                .formLogin(customizer -> customizer
+                        .loginPage("/user/loginPage")
+                        .loginProcessingUrl("/login")
+                        .defaultSuccessUrl("/loginSuccess"))
+                .logout(customizer -> customizer
+                        .logoutSuccessUrl("/"));
         return httpSecurity.build();
     }
     @Bean
Index: src/main/resources/templates/registration.html
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/resources/templates/registration.html b/src/main/resources/templates/registration.html
--- a/src/main/resources/templates/registration.html	(revision 3bb14e1b120aa5df4e884ed09ec2656b3eddaca6)
+++ b/src/main/resources/templates/registration.html	(date 1713089330240)
@@ -50,7 +50,7 @@
                 </div>
                 <div class="title">
                     <h1>Registration</h1>
-                    <p>Have an account? <a href="loginPage.html">Login</a></p>
+                    <p>Have an account? <a href="/user/loginPage">Login</a></p>
                 </div>
                 <span th:if="${msg != null}" th:text="${msg}" style="color: red"></span>
 
